packages/clerk-js/src/ui/common/SSOCallback.tsx

```diff
@@ -24,7 +24,9 @@ export const SSOCallbackCard = (props: HandleOAuthCallbackParams | HandleSamlCal
   React.useEffect(() => {
     let timeoutId: ReturnType<typeof setTimeout>;
     if (__internal_setActiveInProgress !== true) {
-      handleRedirectCallback({ ...props }, navigate).catch(e => {
+      const intent = new URLSearchParams(window.location.search).get('intent');
+      const reloadResource = intent === 'signIn' || intent === 'signUp' ? intent : undefined;
```

Comment:
Can we do this ? Or is `reloadResource:false` not proper ?
```suggestion
      const reloadResource = ['signIn', 'signUp'].includes(intent)
```

Reply:
TypeScript won't narrow from `string` to `'signIn' | 'signUp'` if we use `Array.includes`.